### PR TITLE
Moving from dockercloud to GitHub action + multi-arch support

### PR DESCRIPTION
**Pull request by ilteoood**
#### What is the purpose of this change?

With this PR I would like to move the CI for docker images from dockercloud to GitHub Actions.
Also, I would like to add support for multi-arch images in order to support not only amd64 but also arm, x86... architectures

#### Was the change discussed in an issue or in the forum before?

I don't know

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: .github/workflows/build_publish_docker_image.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    build:
+        runs-on: ubuntu-latest
+        name: Build image job
+        steps:
+            - name: Checkout master
+              uses: actions/checkout@master
+            - name: Build and publish image
+              uses: ilteoood/docker_buildx@master
+              with:
+                tag: latest,beta
```
Comment (ncw) [inline .github/workflows/build_publish_docker_image.yml:20]: How does the tagging work?

Currently we use
- `beta` meaning latest master build
- `latest` meaning latest stable build - this is a build with a git tag on
- `1`  latest in the 1 series
- `1.52`  latest in the 1.52 series
- `1.52.0`  actual version number

So we need to do something different for building betas vs building the `latest`.

Maybe tagging the latest could be a different step?

File: .github/workflows/build_publish_docker_image.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                imageName: rclone/rclone
+                platform: linux/amd64,linux/arm64,linux/ppc64le,linux/s390x,linux/386,linux/arm/v7
+                publish: true
+                dockerHubUser: rclone
+                dockerHubPassword: ${{ secrets.DOCKER_HUB_PASSWORD }}
```
Comment (ncw) [inline .github/workflows/build_publish_docker_image.yml:25]: Presumably I'd need to put my actual secrets in the GitHub actions secrets

File: .github/workflows/build_publish_release_docker_image.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        runs-on: ubuntu-latest
+        name: Build image job
+        steps:
+            - name: Checkout master
+              uses: actions/checkout@master
+            - name: Get the version
+              id: get_version
+              run: echo ::set-output name=VERSION::$(echo $GITHUB_REF | cut -d / -f 3)
+            - name: Build and publish image
+              uses: ilteoood/docker_buildx@master
```
Comment (ncw) [inline .github/workflows/build_publish_release_docker_image.yml:17]: So this is your creation? Very nice :-)

File: .github/workflows/build_publish_release_docker_image.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: Build and publish release image
+on:
+  release:
+    types: [published]
```
Comment (ncw) [inline .github/workflows/build_publish_release_docker_image.yml:4]: What does this `published` release mean?

File: .github/workflows/build_publish_docker_image.yml (hunk 1 above)
Comment (ilteoood) [reply .github/workflows/build_publish_docker_image.yml:20]: This pipeline work as follow: everytime you push or create a release tag, a new version is uploaded to docker hub with tag latest and beta.

Can I assume that everytime that you push on master, I can upload with beta tag?
The rest of the tag should be added only on the creation of a new release. Right?

**Comment by ilteoood** [comment]
Hi @ncw ,
I've updated the pipeline.
With the first file, every time that you commit on master a new image with tag beta is uploaded to docker hub.
Otherwise when you create a new release, an image with 4 tags is built: latest, major, minr and patch version; for example: latest, 1, 1.52 and 1.52.0

Can you integrate it now? I would like to use rclone on my rpi :smile:

File: .github/workflows/build_publish_release_docker_image.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+              uses: actions/checkout@master
+            - name: Get actual patch version
+              id: actual_patch_version
+              run: echo ::set-output name=ACTUAL_PATCH_VERSION::$(echo $GITHUB_REF | cut -d / -f 3 | sed 's/v//g')
+            - name: Get actual minor version
+              id: actual_minor_version
+              run: echo ::set-output name=ACTUAL_MINOR_VERSION::$(echo $GITHUB_REF | cut -d / -f 3 | sed 's/v//g' | cut -d "." -f 1,2)
+            - name: Get actual major version
+              id: actual_major_version
+              run: echo ::set-output name=ACTUAL_MAJOR_VERSION::$(echo $GITHUB_REF | cut -d / -f 3 | sed 's/v//g' | cut -d "." -f 1,2)
```
Comment (ncw) [inline .github/workflows/build_publish_release_docker_image.yml:21]: I think this should probably be `...| cut -d "." -f 1)` not `-f 1,2` ?

File: .github/workflows/build_publish_release_docker_image.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            - name: Build and publish image
+              uses: ilteoood/docker_buildx@master
+              with:
+                tag: latest,${{ steps.actual_patch_version.outputs.ACTUAL_PATCH_VERSION }},${{ steps.actual_minor_version.outputs.ACTUAL_MINOR_VERSION }},${{ steps.actual_major_version.outputs.ACTUAL_MAJOR_VERSION }}
+                imageName: rclone/rclone
+                platform: linux/amd64,linux/arm64,linux/ppc64le,linux/s390x,linux/386,linux/arm/v7
+                publish: true
+                dockerHubUser: rclone
```
Comment (ncw) [inline .github/workflows/build_publish_release_docker_image.yml:29]: I think this should probably come from `${{ secrets.DOCKER_HUB_USER }}` as my user isn't actually called `rclone`!

File: .github/workflows/build_publish_release_docker_image.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+                dockerHubPassword: ${{ secrets.DOCKER_HUB_PASSWORD }}
```
Comment (ncw) [inline .github/workflows/build_publish_release_docker_image.yml:30]: There should probably be a newline on the end of this file and the other?

File: .github/workflows/build_publish_docker_image.yml (hunk 1 above)
Comment (ncw) [reply .github/workflows/build_publish_docker_image.yml]: I think you got the logic correct here :-) Commits to master make a beta tag, the rest are on the creation of a new release.

File: .github/workflows/build_publish_release_docker_image.yml (hunk 5 above)
Comment (ilteoood) [reply .github/workflows/build_publish_release_docker_image.yml:21]: Yes, sorry a wrong copy-past :/

File: .github/workflows/build_publish_release_docker_image.yml (hunk 6 above)
Comment (ilteoood) [reply .github/workflows/build_publish_release_docker_image.yml:29]: done

**Comment by ilteoood** [comment]
Hi @ncw,
I've made all required changes :smile: 
It's your turn!

**Comment by ncw** [comment]
Thank you for doing this :-)

I'm going to pull this into a branch and test it

**Pull request merged**